### This notebook will be for processing and cleaning the subway turnstile dataset for 2019
- Will need to determine all the turnstile differences for entries between current and last value
- Same for exits 
- This can give me a representation of how many people enter/exit station X in a given time period
- Do this for only stations in Manhattan

In [1]:
import json
import pandas as pd

### Explanation of linking the 3 datasets to get useful information:
- turnstile_data is the dataset obtained from: https://data.ny.gov/Transportation/Turnstile-Usage-Data-2019/xfn5-qji9
- This dataset contains all the info in 2019 for all the turnstiles in the MTA system
- station_data is a dataset obtained from: http://web.mta.info/developers/data/nyct/subway/Stations.csv
- This dataset contains information for all the stations in the MTA system
- The problem that occurs is how to link the two datasets together
- lookup_data

In [34]:
turnstile_data = pd.read_csv('Turnstile_Usage_Data__2019.csv', nrows=1000)
turnstile_data.head()

C/A  Unit       SCP         Station Line Name Division        Date  \
0  A033  R170  02-00-05  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
1  A033  R170  02-00-02  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
2  A033  R170  02-06-00  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
3  A033  R170  02-00-03  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
4  A033  R170  02-06-01  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   

       Time Description   Entries  \
0  00:00:00     REGULAR  17538854   
1  00:00:00     REGULAR  14983900   
2  00:00:00     REGULAR    769115   
3  00:00:00     REGULAR   7191422   
4  00:00:00     REGULAR  71047673   

   Exits                                                       
0                                            7031168           
1                                           14554087           
2                                             559221           
3                                            8417203           
4                                           20925389

In [36]:
with open('stations.csv') as file:
    station_data = pd.read_csv(file)
station_data.head()

ogc_fid  station id complex id gtfs stop id division     line  \
0        1         1.0          1          R01      BMT  Astoria   
1        2         2.0          2          R03      BMT  Astoria   
2        3         3.0          3          R04      BMT  Astoria   
3        4         4.0          4          R05      BMT  Astoria   
4        5         5.0          5          R06      BMT  Astoria   

                stop name borough daytime routes structure  gtfs latitude  \
0  Astoria - Ditmars Blvd       Q            N W  Elevated      40.775036   
1            Astoria Blvd       Q            N W  Elevated      40.770258   
2                   30 Av       Q            N W  Elevated      40.766779   
3                Broadway       Q            N W  Elevated      40.761820   
4                   36 Av       Q            N W  Elevated      40.756804   

   gtfs longitude   north direction label south direction label  
0      -73.912034                     NaN             Manhattan  
1      -73.917843            Ditmars Blvd             Manhattan  
2      -73.921479  Astoria - Ditmars Blvd             Manhattan  
3      -73.925508  Astoria - Ditmars Blvd             Manhattan  
4      -73.929575  Astoria - Ditmars Blvd             Manhattan

In [37]:
with open('remote_complex_lookup.csv') as file:
    lookup_data = pd.read_csv(file)
lookup_data.head()

remote   booth  complex_id       station line_name division
0    R001   A060       635.0  WHITEHALL ST        R1      BMT
1    R001   A058       635.0  WHITEHALL ST        R1      BMT
2    R001  R101S       635.0   SOUTH FERRY        R1      IRT
3    R002   A077       628.0     FULTON ST  ACJZ2345      BMT
4    R002   A081       628.0     FULTON ST  ACJZ2345      BMT

#### Drop everything not in Manhattan:

In [21]:
for i in range(len(stations)-17):
    if stations['borough'][i] != 'M' :
        stations.drop(i, inplace=True)

stations.head(20)

ogc_fid  station id complex id gtfs stop id division  \
6          7         7.0        613          R11      BMT   
7          8         8.0          8          R13      BMT   
8          9         9.0          9          R14      BMT   
9         10        10.0         10          R15      BMT   
10        11        11.0        611          R16      BMT   
11        12        12.0        607          R17      BMT   
12        13        13.0         13          R18      BMT   
13        14        14.0         14          R19      BMT   
14        15        15.0        602          R20      BMT   
15        16        16.0         16          R21      BMT   
16        17        17.0         17          R22      BMT   
17        18        18.0        623          R23      BMT   
18        19        19.0        623          Q01      BMT   
19        20        20.0         20          R24      BMT   
20        21        21.0        624          R25      BMT   
21        22        22.0         22          R26      BMT   
22        23        23.0        635          R27      BMT   
101      102       102.0        625          M18      BMT   
102      103       103.0        103          M19      BMT   
103      104       104.0        623          M20      BMT   

                    line           stop name borough daytime routes structure  \
6                Astoria  Lexington Av/59 St       M          N W R    Subway   
7                Astoria          5 Av/59 St       M          N W R    Subway   
8    Broadway - Brighton        57 St - 7 Av       M        N Q R W    Subway   
9    Broadway - Brighton               49 St       M          N R W    Subway   
10   Broadway - Brighton    Times Sq - 42 St       M        N Q R W    Subway   
11   Broadway - Brighton   34 St - Herald Sq       M        N Q R W    Subway   
12   Broadway - Brighton               28 St       M            R W    Subway   
13   Broadway - Brighton               23 St       M            R W    Subway   
14   Broadway - Brighton    14 St - Union Sq       M        N Q R W    Subway   
15   Broadway - Brighton          8 St - NYU       M            R W    Subway   
16   Broadway - Brighton           Prince St       M            R W    Subway   
17              Broadway            Canal St       M            R W    Subway   
18      Manhattan Bridge            Canal St       M            N Q    Subway   
19              Broadway           City Hall       M            R W    Subway   
20              Broadway        Cortlandt St       M            R W    Subway   
21              Broadway           Rector St       M            R W    Subway   
22              Broadway        Whitehall St       M            R W    Subway   
101              Jamaica            Essex St       M          J M Z    Subway   
102              Jamaica              Bowery       M            J Z    Subway   
103              Jamaica            Canal St       M            J Z    Subway   

     gtfs latitude  gtfs longitude north direction label  \
6        40.762660      -73.967258                Queens   
7        40.764811      -73.973347                Queens   
8        40.764664      -73.980658       Uptown & Queens   
9        40.759901      -73.984139       Uptown & Queens   
10       40.754672      -73.986754       Uptown & Queens   
11       40.749567      -73.987950       Uptown & Queens   
12       40.745494      -73.988691       Uptown & Queens   
13       40.741303      -73.989344       Uptown & Queens   
14       40.735736      -73.990568       Uptown & Queens   
15       40.730328      -73.992629       Uptown & Queens   
16       40.724329      -73.997702       Uptown & Queens   
17       40.719527      -74.001775       Uptown & Queens   
18       40.718383      -74.000460       Uptown & Queens   
19       40.713282      -74.006978       Uptown & Queens   
20       40.710668      -74.011029       Uptown & Queens   
21       40.707220      -74.013342       Uptown & Queens   
2

In [6]:
len(stations)

224

### Extract subway names and co-ordinates from overpass turbo json file


In [3]:
#there are 161 stations in this dataset

# Open the geojson file and read it
with open('manSubTurn.geojson') as file:
    data = json.load(file)


subway=pd.DataFrame(columns=['name','short_name','lat','lon','Link to Turnstile Data'])

for i in range(len(data['features'])):
    #Extract 
    name=data['features'][i]['properties']['name']
    try:
        short_n=data['features'][i]['properties']['short_name']
    except:
        short_n='None'
        
    lat=data['features'][i]['geometry']['coordinates'][1]
    lon=data['features'][i]['geometry']['coordinates'][0]
    subway.loc[i]=[name,short_n,lat,lon,'None']
    

In [4]:
subway.tail(20)

name             short_name        lat        lon  \
142       Grand Central Terminal                   None  40.752806 -73.977179   
143                  Marble Hill                   None  40.874924 -73.912702   
144                    Manhattan                   None  41.418606 -87.989013   
145                  59th Street                  59 St  40.762707 -73.967880   
146              Chambers Street            Chambers St  40.714916 -74.007968   
147                Fulton Street              Fulton St  40.710186 -74.007664   
148                  49th Street                  49 St  40.759901 -73.984139   
149    34th Street–Herald Square        34 St–Herald Sq  40.749719 -73.987823   
150                  23rd Street                  23 St  40.742853 -73.992834   
151                 Union Square               Union Sq  40.734603 -73.990360   
152     34th Street–Hudson Yards     34 St–Hudson Yards  40.755913 -74.001900   
153  59th Street–Columbus Circle  59 St–Columbus Circle  40.768381 -73.981916   
154                   8th Avenue                   8 Av  40.739875 -74.002738   
155     42nd Street–Times Square         42 St–Times Sq  40.755286 -73.987216   
156                 Times Square               Times Sq  40.755592 -73.985839   
157                   6th Avenue                   6 Av  40.737459 -73.997068   
158                 Canal Street               Canal St  40.718620 -74.000768   
159                 168th Street                 168 St  40.840508 -73.940226   
160                Fulton Street              Fulton St  40.710364 -74.009570   
161                Fulton Street              Fulton St  40.709426 -74.006586   

    Link to Turnstile Data  
142                   None  
143                   None  
144                   None  
145                   None  
146                   None  
147                   None  
148                   None  
149                   None  
150                   None  
151                   None  
152                   None  
153                   None  
154                   None  
155                   None  
156                   None  
157                   None  
158                   None  
159                   None  
160                   None  
161                   None

In [5]:
#remove duplicates? depends on how granular the data for 2019 is

### Now import the turnstile data for 2019

In [23]:
with open('Turnstile_Usage_Data__2019.csv') as file:
    turn_data = pd.read_csv(file)

turn_data.head(20)

C/A  Unit       SCP         Station Line Name Division        Date  \
0   A033  R170  02-00-05  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
1   A033  R170  02-00-02  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
2   A033  R170  02-06-00  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
3   A033  R170  02-00-03  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
4   A033  R170  02-06-01  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
5   A033  R170  02-00-01  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
6   A033  R170  02-00-00  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
7   A033  R170  02-00-04  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
8   A034  R170  03-00-02  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
9   A034  R170  03-03-00  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
10  A034  R170  03-05-00  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
11  A034  R170  03-03-02  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
12  A034  R170  03-05-01  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
13  A034  R170  03-03-01  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
14  A034  R170  03-05-02  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
15  A034  R170  03-00-01  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
16  A034  R170  03-05-03  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
17  A034  R170  03-00-00  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
18  A035  R170  00-00-02  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   
19  A035  R170  00-00-04  14 ST-UNION SQ  LNQR456W      BMT  12/27/2019   

        Time Description   Entries  \
0   00:00:00     REGULAR  17538854   
1   00:00:00     REGULAR  14983900   
2   00:00:00     REGULAR    769115   
3   00:00:00     REGULAR   7191422   
4   00:00:00     REGULAR  71047673   
5   00:00:00     REGULAR   2430093   
6   00:00:00     REGULAR    271981   
7   00:00:00     REGULAR   6483080   
8   00:00:00     REGULAR   4632187   
9   00:00:00     REGULAR     69926   
10  00:00:00     REGULAR    187295   
11  00:00:00     REGULAR  13154638   
12  00:00:00     REGULAR    169531   
13  00:00:00     REGULAR   6553678   
14  00:00:00     REGULAR    198226   
15  00:00:00     REGULAR   1007426   
16  00:00:00     REGULAR    226289   
17  00:00:00     REGULAR   2167998   
18  00:00:00     REGULAR    993554   
19  00:00:00     REGULAR   3119391   

    Exits                                                       
0                                             7031168           
1                                            14554087           
2                                              559221           
3                                             8417203           
4                                            20925389           
5                                             2921770           
6                                              828662           
7                                             4945335           
8                                             3269462           
9                                              219187           
10                                             129140           
11                                            1446679           
12                                             203256           
13                                            3942370           
14                                             260680           
15                                            2330092           
16                                             172333           
17                                           16966778           
18                                            2432341           
19                                             554073

In [24]:
#find how many unique subway names there are
turn_data['Station'].nunique()

379

#### Columns of interest
- Station, this is the station name which I will compare with Overpass turbo name data to only have manhattan turnstile data
- Date, needed for determing day of the week and time of year
- Time, needed for determing time of day
- Entries, needed for determing how many people entered the station, useful for busyness
- Exits, needed for determing how many people exited the station, useful for busyness
- Might also need to include individual turnstile data to see where people are entering/exiting the station and for comparing that turnstile to its previous value to see the difference

### Drop the rows that are not in Manhattan
#### Change overpass turbo names to match turnstile names

In [8]:
#capitalize all overpass turbo names
subway['name'] = subway['name'].str.upper()
subway['short_name'] = subway['short_name'].str.upper()
subway.head(20)

name           short_name        lat        lon  \
0                     96TH STREET                96 ST  40.794379 -73.972000   
1                   BOWLING GREEN                 NONE  40.704509 -74.014095   
2                     SOUTH FERRY                 NONE  40.701710 -74.013149   
3                     86TH STREET                86 ST  40.779494 -73.955529   
4                     28TH STREET                28 ST  40.743314 -73.984075   
5        14TH STREET–UNION SQUARE       14 ST–UNION SQ  40.735761 -73.990649   
6                     28TH STREET                28 ST  40.745494 -73.988691   
7                    CANAL STREET             CANAL ST  40.718092 -73.999892   
8       BROADWAY–LAFAYETTE STREET   B’WAY–LAFAYETTE ST  40.725297 -73.996204   
9                      5TH AVENUE                 5 AV  40.753743 -73.981900   
10      42ND STREET–GRAND CENTRAL  42 ST–GRAND CENTRAL  40.751772 -73.976825   
11         5TH AVENUE–59TH STREET           5 AV–59 ST  40.764827 -73.972675   
12            HARLEM–148TH STREET        HARLEM–148 ST  40.823712 -73.936101   
13  163RD STREET–AMSTERDAM AVENUE  163 ST–AMSTERDAM AV  40.836945 -73.940126   
14                  RECTOR STREET            RECTOR ST  40.707363 -74.013850   
15                  WTC CORTLANDT                 NONE  40.711115 -74.012270   
16                CHAMBERS STREET          CHAMBERS ST  40.715508 -74.009245   
17                    WALL STREET              WALL ST  40.706724 -74.009183   
18                  EAST BROADWAY        EAST BROADWAY  40.713715 -73.990173   
19                   GRAND STREET             GRAND ST  40.718162 -73.993784   

   Link to Turnstile Data  
0                    None  
1                    None  
2                    None  
3                    None  
4                    None  
5                    None  
6                    None  
7                    None  
8                    None  
9                    None  
10                   None  
11                   None  
12                   None  
13                   None  
14                   None  
15                   None  
16                   None  
17                   None  
18                   None  
19                   None

In [9]:
# Now search for the short name in the turnstile data
# If it is not there, search for the long name

stations_to_extract_from_turnstile = []
change_name_stations_to_extract_from_turnstile = []
for i in range(len(subway)):
    name = subway.iloc[i]['name']
    short_name = subway.iloc[i]['short_name']
    if short_name in turn_data['Station'].values:
        stations_to_extract_from_turnstile.append(short_name)
        subway.loc[i,'Link to Turnstile Data'] = short_name
    elif name in turn_data['Station'].values:
        stations_to_extract_from_turnstile.append(name)
        subway.loc[i,'Link to Turnstile Data'] = name
    else:
        change_name_stations_to_extract_from_turnstile.append('None for '+name)
        

print(stations_to_extract_from_turnstile)

['96 ST', 'BOWLING GREEN', 'SOUTH FERRY', '86 ST', '28 ST', '28 ST', 'CANAL ST', 'RECTOR ST', 'CHAMBERS ST', 'WALL ST', 'EAST BROADWAY', 'GRAND ST', 'BOWERY', 'BROAD ST', '2 AV', '51 ST', '7 AV', 'PRINCE ST', 'CANAL ST', '23 ST', 'ASTOR PL', 'BLEECKER ST', 'SPRING ST', '33 ST', '1 AV', 'CANAL ST', '14 ST', '3 AV', 'SPRING ST', '23 ST', '28 ST', '50 ST', 'HOUSTON ST', '18 ST', 'FRANKLIN ST', '72 ST', '77 ST', '96 ST', 'DYCKMAN ST', '125 ST', '125 ST', '135 ST', '145 ST', '79 ST', '86 ST', '110 ST', '116 ST', '103 ST', '125 ST', '155 ST', '116 ST', '103 ST', '116 ST', '14 ST', '57 ST', 'CITY HALL', '155 ST', 'CHAMBERS ST', 'PARK PLACE', 'CORTLANDT ST', 'RECTOR ST', '23 ST', 'DYCKMAN ST', '168 ST', '181 ST', '191 ST', '207 ST', '215 ST', '145 ST', '157 ST', '103 ST', '72 ST', '86 ST', '96 ST', 'CANAL ST', '9TH STREET', '14 ST', '14 ST', '33 ST', 'FULTON ST', 'WALL ST', '125 ST', '145 ST', '181 ST', '190 ST', '23 ST', '23 ST', 'CHRISTOPHER ST', '135 ST', '86 ST', '96 ST', '72 ST', 'CANAL S

In [10]:
subway.head(20)

name           short_name        lat        lon  \
0                     96TH STREET                96 ST  40.794379 -73.972000   
1                   BOWLING GREEN                 NONE  40.704509 -74.014095   
2                     SOUTH FERRY                 NONE  40.701710 -74.013149   
3                     86TH STREET                86 ST  40.779494 -73.955529   
4                     28TH STREET                28 ST  40.743314 -73.984075   
5        14TH STREET–UNION SQUARE       14 ST–UNION SQ  40.735761 -73.990649   
6                     28TH STREET                28 ST  40.745494 -73.988691   
7                    CANAL STREET             CANAL ST  40.718092 -73.999892   
8       BROADWAY–LAFAYETTE STREET   B’WAY–LAFAYETTE ST  40.725297 -73.996204   
9                      5TH AVENUE                 5 AV  40.753743 -73.981900   
10      42ND STREET–GRAND CENTRAL  42 ST–GRAND CENTRAL  40.751772 -73.976825   
11         5TH AVENUE–59TH STREET           5 AV–59 ST  40.764827 -73.972675   
12            HARLEM–148TH STREET        HARLEM–148 ST  40.823712 -73.936101   
13  163RD STREET–AMSTERDAM AVENUE  163 ST–AMSTERDAM AV  40.836945 -73.940126   
14                  RECTOR STREET            RECTOR ST  40.707363 -74.013850   
15                  WTC CORTLANDT                 NONE  40.711115 -74.012270   
16                CHAMBERS STREET          CHAMBERS ST  40.715508 -74.009245   
17                    WALL STREET              WALL ST  40.706724 -74.009183   
18                  EAST BROADWAY        EAST BROADWAY  40.713715 -73.990173   
19                   GRAND STREET             GRAND ST  40.718162 -73.993784   

   Link to Turnstile Data  
0                   96 ST  
1           BOWLING GREEN  
2             SOUTH FERRY  
3                   86 ST  
4                   28 ST  
5                    None  
6                   28 ST  
7                CANAL ST  
8                    None  
9                    None  
10                   None  
11                   None  
12                   None  
13                   None  
14              RECTOR ST  
15                   None  
16            CHAMBERS ST  
17                WALL ST  
18          EAST BROADWAY  
19               GRAND ST

In [11]:
print(change_name_stations_to_extract_from_turnstile)
#will need to change these names manually

['None for 14TH STREET–UNION SQUARE', 'None for BROADWAY–LAFAYETTE STREET', 'None for 5TH AVENUE', 'None for 42ND STREET–GRAND CENTRAL', 'None for 5TH AVENUE–59TH STREET', 'None for HARLEM–148TH STREET', 'None for 163RD STREET–AMSTERDAM AVENUE', 'None for WTC CORTLANDT', 'None for WEST 4TH STREET–WASHINGTON SQUARE', 'None for ESSEX STREET', 'None for LEXINGTON AVENUE–63RD STREET', 'None for 42ND STREET–BRYANT PARK', 'None for LEXINGTON AVENUE–59TH STREET', 'None for 5TH AVENUE–53RD STREET', 'None for 57TH STREET–7TH AVENUE', 'None for 8TH STREET–NEW YORK UNIVERSITY', 'None for 42ND STREET–PORT AUTHORITY BUS TERMINAL', 'None for ROOSEVELT ISLAND', 'None for NEW YORK PENN STATION', 'None for 66TH STREET–LINCOLN CENTER', 'None for WORLD TRADE CENTER', 'None for CENTRAL PARK NORTH–110TH STREET', 'None for BROOKLYN BRIDGE–CITY HALL', 'None for 42ND STREET–TIMES SQUARE', 'None for MARBLE HILL–225TH STREET', 'None for 137TH STREET–CITY COLLEGE', 'None for CATHEDRAL PARKWAY–110TH STREET', 'Non

In [12]:
print(len(change_name_stations_to_extract_from_turnstile))

56


In [13]:
change_name_stations_to_extract_from_turnstile.append('14 ST-UNION SQ')
change_name_stations_to_extract_from_turnstile.append('B\'WAY-LAFAYETTE')
change_name_stations_to_extract_from_turnstile.append('5 AVE')
change_name_stations_to_extract_from_turnstile.append('GRD CNTRL-42 ST')
change_name_stations_to_extract_from_turnstile.append('5 AV/59 ST')
change_name_stations_to_extract_from_turnstile.append('HARLEM 148 ST')
change_name_stations_to_extract_from_turnstile.append('163 ST-AMSTERDM')
change_name_stations_to_extract_from_turnstile.append('WTC-CORTLANDT')
change_name_stations_to_extract_from_turnstile.append('W 4 ST-WASH SQ')
change_name_stations_to_extract_from_turnstile.append('DELANCEY/ESSEX')
change_name_stations_to_extract_from_turnstile.append('LEXINGTON AV/63')
change_name_stations_to_extract_from_turnstile.append('42 ST-BRYANT PK')
change_name_stations_to_extract_from_turnstile.append('None')
change_name_stations_to_extract_from_turnstile.append('5 AV/53 ST')
change_name_stations_to_extract_from_turnstile.append('57 ST-7 AV')
change_name_stations_to_extract_from_turnstile.append('8 ST-NYU')
change_name_stations_to_extract_from_turnstile.append('None as bus terminal')
change_name_stations_to_extract_from_turnstile.append('ROOSEVELT ISLND')
change_name_stations_to_extract_from_turnstile.append('34 ST-PENN STA')
change_name_stations_to_extract_from_turnstile.append('66 ST-LINCOLN')
change_name_stations_to_extract_from_turnstile.append('WORLD TRADE CTR')
change_name_stations_to_extract_from_turnstile.append('CENTRAL PK N110')
change_name_stations_to_extract_from_turnstile.append('BROOKLYN BRIDGE')
change_name_stations_to_extract_from_turnstile.append('TIMES SQ-42 ST')
change_name_stations_to_extract_from_turnstile.append('MARBLE HILL-225')
change_name_stations_to_extract_from_turnstile.append('137 ST CITY COL')
change_name_stations_to_extract_from_turnstile.append('CATHEDRAL PKWY')
change_name_stations_to_extract_from_turnstile.append('116 ST-COLUMBIA')
change_name_stations_to_extract_from_turnstile.append('34 ST-PENN STA')
change_name_stations_to_extract_from_turnstile.append('TIMES SQ-42 ST')
change_name_stations_to_extract_from_turnstile.append('CHRISTOPHER ST')
change_name_stations_to_extract_from_turnstile.append('CATHEDRAL PKWY')
change_name_stations_to_extract_from_turnstile.append('68ST-HUNTER CO')
change_name_stations_to_extract_from_turnstile.append('34 ST-PENN STA')
change_name_stations_to_extract_from_turnstile.append('LEXINGTON AV/53')
change_name_stations_to_extract_from_turnstile.append('47-50 STS ROCK')
change_name_stations_to_extract_from_turnstile.append('DELANCEY/ESSEX')
change_name_stations_to_extract_from_turnstile.append('34 ST-HERALD SQ')
change_name_stations_to_extract_from_turnstile.append('81 ST-MUSEUM')
change_name_stations_to_extract_from_turnstile.append('59 ST COLUMBUS')
change_name_stations_to_extract_from_turnstile.append('WHITEHALL S-FRY')
change_name_stations_to_extract_from_turnstile.append('WORLD TRADE CTR')
change_name_stations_to_extract_from_turnstile.append('14 ST-UNION SQ')
change_name_stations_to_extract_from_turnstile.append('GRD CNTRL-42 ST')
change_name_stations_to_extract_from_turnstile.append('GRD CNTRL-42 ST')
change_name_stations_to_extract_from_turnstile.append('INWOOD-207 ST')
change_name_stations_to_extract_from_turnstile.append('125 ST')
change_name_stations_to_extract_from_turnstile.append('MARBLE HILL-225')
change_name_stations_to_extract_from_turnstile.append('GRD CNTRL-42 ST')
change_name_stations_to_extract_from_turnstile.append('34 ST-HERALD SQ')
change_name_stations_to_extract_from_turnstile.append('14 ST-UNION SQ')
change_name_stations_to_extract_from_turnstile.append('34 ST-HUDSON YD')
change_name_stations_to_extract_from_turnstile.append('59 ST COLUMBUS')
change_name_stations_to_extract_from_turnstile.append('TIMES SQ-42 ST')
change_name_stations_to_extract_from_turnstile.append('TIMES SQ-42 ST')
change_name_stations_to_extract_from_turnstile.append('TIMES SQ-42 ST')

In [14]:
len(change_name_stations_to_extract_from_turnstile)

112

- Missing for LEXINGTON AVENUE–59TH STREET

- Have a triple of penn station 34 st
- Duplicate of TIMES SQ-42 ST
- Duplicate of CATHEDRAL PKWY
- Duplicate of DELANCEY/ESSEX
- Duplicate of WORLD TRADE CTR
- Duplicate of 14 ST-UNION SQ
- Quadruple of GRD CNTRL-42 ST
- Duplicate of 125 ST
- Duplicate of MARBLE HILL-225
- 



In [15]:
stations_to_extract_from_turnstile = []
j=56
#change_name_stations_to_extract_from_turnstile = []
for i in range(len(subway)):
    name = subway.iloc[i]['name']
    short_name = subway.iloc[i]['short_name']
    if short_name in turn_data['Station'].values:
        stations_to_extract_from_turnstile.append(short_name)
        subway.loc[i,'Link to Turnstile Data'] = short_name
    elif name in turn_data['Station'].values:
        stations_to_extract_from_turnstile.append(name)
        subway.loc[i,'Link to Turnstile Data'] = name
    else:
        subway.loc[i,'Link to Turnstile Data'] = change_name_stations_to_extract_from_turnstile[j]
        j=j+1

subway.head(20)

name           short_name        lat        lon  \
0                     96TH STREET                96 ST  40.794379 -73.972000   
1                   BOWLING GREEN                 NONE  40.704509 -74.014095   
2                     SOUTH FERRY                 NONE  40.701710 -74.013149   
3                     86TH STREET                86 ST  40.779494 -73.955529   
4                     28TH STREET                28 ST  40.743314 -73.984075   
5        14TH STREET–UNION SQUARE       14 ST–UNION SQ  40.735761 -73.990649   
6                     28TH STREET                28 ST  40.745494 -73.988691   
7                    CANAL STREET             CANAL ST  40.718092 -73.999892   
8       BROADWAY–LAFAYETTE STREET   B’WAY–LAFAYETTE ST  40.725297 -73.996204   
9                      5TH AVENUE                 5 AV  40.753743 -73.981900   
10      42ND STREET–GRAND CENTRAL  42 ST–GRAND CENTRAL  40.751772 -73.976825   
11         5TH AVENUE–59TH STREET           5 AV–59 ST  40.764827 -73.972675   
12            HARLEM–148TH STREET        HARLEM–148 ST  40.823712 -73.936101   
13  163RD STREET–AMSTERDAM AVENUE  163 ST–AMSTERDAM AV  40.836945 -73.940126   
14                  RECTOR STREET            RECTOR ST  40.707363 -74.013850   
15                  WTC CORTLANDT                 NONE  40.711115 -74.012270   
16                CHAMBERS STREET          CHAMBERS ST  40.715508 -74.009245   
17                    WALL STREET              WALL ST  40.706724 -74.009183   
18                  EAST BROADWAY        EAST BROADWAY  40.713715 -73.990173   
19                   GRAND STREET             GRAND ST  40.718162 -73.993784   

   Link to Turnstile Data  
0                   96 ST  
1           BOWLING GREEN  
2             SOUTH FERRY  
3                   86 ST  
4                   28 ST  
5          14 ST-UNION SQ  
6                   28 ST  
7                CANAL ST  
8         B'WAY-LAFAYETTE  
9                   5 AVE  
10        GRD CNTRL-42 ST  
11             5 AV/59 ST  
12          HARLEM 148 ST  
13        163 ST-AMSTERDM  
14              RECTOR ST  
15          WTC-CORTLANDT  
16            CHAMBERS ST  
17                WALL ST  
18          EAST BROADWAY  
19               GRAND ST

In [16]:
#show the entire data frame
#pd.set_option('display.max_rows', None)
#subway

In [17]:
#remove any row that has a duplicate value in the link to turnstile data column
subway = subway.drop_duplicates(subset=['Link to Turnstile Data'], keep='first')
len(subway)

96

### Now have 96 subway locations in Manhattan that we can reference in the turnstile data


#### is is it 146 from station_link?

In [25]:
#station_link['station']
man_turn_data = pd.merge(stations_link, turn_data, left_on='station', right_on='Station')
man_turn_data.head(20)

station line_names division    borough subregion    C/A  Unit       SCP  \
0     1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-03-00   
1     1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-00-00   
2     1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-03-01   
3     1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-03-02   
4     1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-00-01   
5     1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-01   
6     1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-00   
7     1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-04   
8     1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-03   
9     1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-02   
10    1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-03-02   
11    1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-00-01   
12    1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-03-01   
13    1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-00-00   
14    1 AV          L      BMT  Manhattan  14 to 42   H007  R248  00-03-00   
15    1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-01   
16    1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-04   
17    1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-03   
18    1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-02   
19    1 AV          L      BMT  Manhattan  14 to 42  H007A  R248  02-03-00   

   Station Line Name Division        Date      Time Description    Entries  \
0     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR  370818021   
1     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR   15302393   
2     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR    2569788   
3     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR    6577135   
4     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR   60949253   
5     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR      14996   
6     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR       3835   
7     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR      84669   
8     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR      50430   
9     1 AV         L      BMT  12/27/2019  03:00:00     REGULAR      32577   
10    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR    6577154   
11    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR   60949311   
12    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR    2569795   
13    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR   15302456   
14    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR  370818038   
15    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR      14999   
16    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR      84706   
17    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR      50447   
18    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR      32584   
19    1 AV         L      BMT  12/27/2019  07:00:00     REGULAR       3835   

    Exits                                                       
0                                           387993138           
1                                            17102988           
2                                             1109743           
3                                              553888           
4                                            38064126           
5                                              117639           
6                                              192369           
7                                               12012           
8                                               29628           
9                                               62861           
10        

In [18]:
#trim the csv data to a new file containg the stations in manhattan
#this will be the data we use for the rest of the project

man_turn_data = pd.DataFrame(columns=turn_data.columns)  # Initialize an empty DataFrame with the same columns as turn_data
'''
for i in range(len(subway)):
    station_to_find=subway['Link to Turnstile Data'][i]
    print(station_to_find)
    for j in range(len(turn_data)):
        if turn_data['Station'][j] == station_to_find:
            man_turn_data = man_turn_data.append(turn_data.iloc[j], ignore_index=True)

man_turn_data
'''

"\nfor i in range(len(subway)):\n    station_to_find=subway['Link to Turnstile Data'][i]\n    print(station_to_find)\n    for j in range(len(turn_data)):\n        if turn_data['Station'][j] == station_to_find:\n            man_turn_data = man_turn_data.append(turn_data.iloc[j], ignore_index=True)\n\nman_turn_data\n"

In [19]:
man_turn_data = pd.merge(subway, turn_data, left_on='Link to Turnstile Data', right_on='Station')


In [20]:
man_turn_data.head(10)

name short_name        lat     lon Link to Turnstile Data    C/A  \
0  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
1  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
2  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
3  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
4  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
5  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
6  96TH STREET      96 ST  40.794379 -73.972                  96 ST  R168A   
7  96TH STREET      96 ST  40.794379 -73.972                  96 ST   R169   
8  96TH STREET      96 ST  40.794379 -73.972                  96 ST   R169   
9  96TH STREET      96 ST  40.794379 -73.972                  96 ST   R169   

   Unit       SCP Station Line Name Division        Date      Time  \
0  R168  00-00-02   96 ST       123      IRT  12/27/2019  00:00:00   
1  R168  00-02-00   96 ST       123      IRT  12/27/2019  00:00:00   
2  R168  00-03-02   96 ST       123      IRT  12/27/2019  00:00:00   
3  R168  00-00-00   96 ST       123      IRT  12/27/2019  00:00:00   
4  R168  00-00-01   96 ST       123      IRT  12/27/2019  00:00:00   
5  R168  00-03-01   96 ST       123      IRT  12/27/2019  00:00:00   
6  R168  00-03-00   96 ST       123      IRT  12/27/2019  00:00:00   
7  R168  01-00-01   96 ST       123      IRT  12/27/2019  00:00:00   
8  R168  01-03-02   96 ST       123      IRT  12/27/2019  00:00:00   
9  R168  01-05-01   96 ST       123      IRT  12/27/2019  00:00:00   

  Description   Entries  \
0     REGULAR   7050714   
1     REGULAR   6592911   
2     REGULAR  11982395   
3     REGULAR   1875264   
4     REGULAR   6189156   
5     REGULAR   4028999   
6     REGULAR   4580292   
7     REGULAR   1620619   
8     REGULAR   9245121   
9     REGULAR         0   

   Exits                                                       
0                                            3195113           
1                                            7757191           
2                                           11692946           
3                                            1933375           
4                                            2545523           
5                                            5063535           
6                                           10195272           
7                                             988385           
8                                            8843365           
9                                               1607

In [21]:
len(man_turn_data)
#4.5 million rows, which is 5.6 less than the original 


4576249

### feature selection


In [22]:
#lets drop:
#short_name, link to turnstile data, Unit,  Line Name, Division
# if desc is recover aud disregard those rows?
man_turn_data.drop(['short_name','Link to Turnstile Data','Unit','Line Name','Division'], axis=1, inplace=True)

In [23]:
man_turn_data.head(20)

name        lat     lon    C/A       SCP Station        Date  \
0   96TH STREET  40.794379 -73.972  R168A  00-00-02   96 ST  12/27/2019   
1   96TH STREET  40.794379 -73.972  R168A  00-02-00   96 ST  12/27/2019   
2   96TH STREET  40.794379 -73.972  R168A  00-03-02   96 ST  12/27/2019   
3   96TH STREET  40.794379 -73.972  R168A  00-00-00   96 ST  12/27/2019   
4   96TH STREET  40.794379 -73.972  R168A  00-00-01   96 ST  12/27/2019   
5   96TH STREET  40.794379 -73.972  R168A  00-03-01   96 ST  12/27/2019   
6   96TH STREET  40.794379 -73.972  R168A  00-03-00   96 ST  12/27/2019   
7   96TH STREET  40.794379 -73.972   R169  01-00-01   96 ST  12/27/2019   
8   96TH STREET  40.794379 -73.972   R169  01-03-02   96 ST  12/27/2019   
9   96TH STREET  40.794379 -73.972   R169  01-05-01   96 ST  12/27/2019   
10  96TH STREET  40.794379 -73.972   R169  01-07-00   96 ST  12/27/2019   
11  96TH STREET  40.794379 -73.972   R169  01-00-04   96 ST  12/27/2019   
12  96TH STREET  40.794379 -73.972   R169  01-03-01   96 ST  12/27/2019   
13  96TH STREET  40.794379 -73.972   R169  01-03-05   96 ST  12/27/2019   
14  96TH STREET  40.794379 -73.972   R169  01-00-00   96 ST  12/27/2019   
15  96TH STREET  40.794379 -73.972   R169  01-00-03   96 ST  12/27/2019   
16  96TH STREET  40.794379 -73.972   R169  01-03-03   96 ST  12/27/2019   
17  96TH STREET  40.794379 -73.972   R169  01-00-02   96 ST  12/27/2019   
18  96TH STREET  40.794379 -73.972   R169  01-07-01   96 ST  12/27/2019   
19  96TH STREET  40.794379 -73.972   R169  01-03-00   96 ST  12/27/2019   

        Time Description   Entries  \
0   00:00:00     REGULAR   7050714   
1   00:00:00     REGULAR   6592911   
2   00:00:00     REGULAR  11982395   
3   00:00:00     REGULAR   1875264   
4   00:00:00     REGULAR   6189156   
5   00:00:00     REGULAR   4028999   
6   00:00:00     REGULAR   4580292   
7   00:00:00     REGULAR   1620619   
8   00:00:00     REGULAR   9245121   
9   00:00:00     REGULAR         0   
10  00:00:00     REGULAR  67110761   
11  00:00:00     REGULAR    754268   
12  00:00:00     REGULAR   1467020   
13  00:00:00     REGULAR  22525135   
14  00:00:00     REGULAR   1789165   
15  00:00:00     REGULAR   3219007   
16  00:00:00     REGULAR  10599617   
17  00:00:00     REGULAR    562125   
18  00:00:00     REGULAR    131072   
19  00:00:00     REGULAR   3983668   

    Exits                                                       
0                                             3195113           
1                                             7757191           
2                                            11692946           
3                                             1933375           
4                                             2545523           
5                                             5063535           
6                                            10195272           
7                                              988385           
8                                             8843365           
9                                                1607           
10                                           16777216           
11                                            1396363           
12                                            3645685           
13                                           14692689           
14                                            2439123           
15                                            3508799           
16                                            4196820           
17                                             408363           
18                                              66532           
19                                           20767239

In [24]:
#group alphabetically by station Date and Time
man_turn_data.sort_values(by=['name'], inplace=True)
man_turn_data.head(20)

name        lat        lon   C/A       SCP Station  \
3007920  103RD STREET  40.790298 -73.947624  R252  00-00-01  103 ST   
3010850  103RD STREET  40.790298 -73.947624  R170  00-00-02  103 ST   
3010851  103RD STREET  40.790298 -73.947624  R170  00-00-01  103 ST   
3010852  103RD STREET  40.790298 -73.947624  R170  00-03-02  103 ST   
3010853  103RD STREET  40.790298 -73.947624  R170  00-00-00  103 ST   
3010854  103RD STREET  40.790298 -73.947624  R170  00-00-00  103 ST   
3010855  103RD STREET  40.790298 -73.947624  R170  00-03-01  103 ST   
3010856  103RD STREET  40.790298 -73.947624  R170  00-03-01  103 ST   
3010857  103RD STREET  40.790298 -73.947624  R170  00-00-01  103 ST   
3010858  103RD STREET  40.790298 -73.947624  R170  00-03-02  103 ST   
3010859  103RD STREET  40.790298 -73.947624  R170  00-00-02  103 ST   
3010860  103RD STREET  40.790298 -73.947624  R170  00-03-00  103 ST   
3010861  103RD STREET  40.790298 -73.947624  R170  00-03-00  103 ST   
3010862  103RD STREET  40.790298 -73.947624  R252  00-00-00  103 ST   
3010863  103RD STREET  40.790298 -73.947624  R252  00-03-02  103 ST   
3010864  103RD STREET  40.790298 -73.947624  R252  00-03-02  103 ST   
3010849  103RD STREET  40.790298 -73.947624  N037  00-00-00  103 ST   
3010848  103RD STREET  40.790298 -73.947624  N037  00-00-01  103 ST   
3010847  103RD STREET  40.790298 -73.947624  N037  00-00-01  103 ST   
3010846  103RD STREET  40.790298 -73.947624  N037  00-00-02  103 ST   

               Date      Time Description   Entries  \
3007920  11/17/2019  16:00:00     REGULAR   4691357   
3010850  10/18/2019  13:00:00     REGULAR  16673625   
3010851  10/18/2019  13:00:00     REGULAR  13639657   
3010852  10/18/2019  13:00:00     REGULAR  11693983   
3010853  10/18/2019  13:00:00     REGULAR   2355096   
3010854  10/18/2019  13:00:00     REGULAR   2355096   
3010855  10/18/2019  13:00:00     REGULAR  12771924   
3010856  10/18/2019  13:00:00     REGULAR  12771924   
3010857  10/18/2019  13:00:00     REGULAR  13639657   
3010858  10/18/2019  13:00:00     REGULAR  11693983   
3010859  10/18/2019  13:00:00     REGULAR  16673625   
3010860  10/18/2019  13:00:00     REGULAR   8457061   
3010861  10/18/2019  13:00:00     REGULAR   8457061   
3010862  10/18/2019  13:00:00     REGULAR    273991   
3010863  10/18/2019  13:00:00     REGULAR  25677885   
3010864  10/18/2019  13:00:00     REGULAR  25677885   
3010849  10/18/2019  13:00:00     REGULAR  14444135   
3010848  10/18/2019  13:00:00     REGULAR  10994822   
3010847  10/18/2019  13:00:00     REGULAR  10994822   
3010846  10/18/2019  13:00:00     REGULAR   5100839   

         Exits                                                       
3007920                                            1945485           
3010850                                            4540530           
3010851                                            3993107           
3010852                                           10975178           
3010853                                             925048           
3010854                                             925048           
3010855                                           10179811           
3010856                                           10179811           
3010857                                            3993107           
3010858                                           10975178           
3010859                                            4540530           
3010860                                            4755459           
3010861                                            4755459           
3010862                                             197583           
3010863                                           44108376           
3010864                                           44108376           
3010849                                           11509079           
3010848                                            6061830           
3010847                                

In [25]:
#convert the date and time to unix timestamps
man_turn_data['DateTime'] = pd.to_datetime(man_turn_data['Date'] + ' ' + man_turn_data['Time'])

man_turn_data['UnixTimestamp'] = man_turn_data['DateTime'].apply(lambda x: x.timestamp())
man_turn_data.head(20)


name        lat        lon   C/A       SCP Station  \
3007920  103RD STREET  40.790298 -73.947624  R252  00-00-01  103 ST   
3010850  103RD STREET  40.790298 -73.947624  R170  00-00-02  103 ST   
3010851  103RD STREET  40.790298 -73.947624  R170  00-00-01  103 ST   
3010852  103RD STREET  40.790298 -73.947624  R170  00-03-02  103 ST   
3010853  103RD STREET  40.790298 -73.947624  R170  00-00-00  103 ST   
3010854  103RD STREET  40.790298 -73.947624  R170  00-00-00  103 ST   
3010855  103RD STREET  40.790298 -73.947624  R170  00-03-01  103 ST   
3010856  103RD STREET  40.790298 -73.947624  R170  00-03-01  103 ST   
3010857  103RD STREET  40.790298 -73.947624  R170  00-00-01  103 ST   
3010858  103RD STREET  40.790298 -73.947624  R170  00-03-02  103 ST   
3010859  103RD STREET  40.790298 -73.947624  R170  00-00-02  103 ST   
3010860  103RD STREET  40.790298 -73.947624  R170  00-03-00  103 ST   
3010861  103RD STREET  40.790298 -73.947624  R170  00-03-00  103 ST   
3010862  103RD STREET  40.790298 -73.947624  R252  00-00-00  103 ST   
3010863  103RD STREET  40.790298 -73.947624  R252  00-03-02  103 ST   
3010864  103RD STREET  40.790298 -73.947624  R252  00-03-02  103 ST   
3010849  103RD STREET  40.790298 -73.947624  N037  00-00-00  103 ST   
3010848  103RD STREET  40.790298 -73.947624  N037  00-00-01  103 ST   
3010847  103RD STREET  40.790298 -73.947624  N037  00-00-01  103 ST   
3010846  103RD STREET  40.790298 -73.947624  N037  00-00-02  103 ST   

               Date      Time Description   Entries  \
3007920  11/17/2019  16:00:00     REGULAR   4691357   
3010850  10/18/2019  13:00:00     REGULAR  16673625   
3010851  10/18/2019  13:00:00     REGULAR  13639657   
3010852  10/18/2019  13:00:00     REGULAR  11693983   
3010853  10/18/2019  13:00:00     REGULAR   2355096   
3010854  10/18/2019  13:00:00     REGULAR   2355096   
3010855  10/18/2019  13:00:00     REGULAR  12771924   
3010856  10/18/2019  13:00:00     REGULAR  12771924   
3010857  10/18/2019  13:00:00     REGULAR  13639657   
3010858  10/18/2019  13:00:00     REGULAR  11693983   
3010859  10/18/2019  13:00:00     REGULAR  16673625   
3010860  10/18/2019  13:00:00     REGULAR   8457061   
3010861  10/18/2019  13:00:00     REGULAR   8457061   
3010862  10/18/2019  13:00:00     REGULAR    273991   
3010863  10/18/2019  13:00:00     REGULAR  25677885   
3010864  10/18/2019  13:00:00     REGULAR  25677885   
3010849  10/18/2019  13:00:00     REGULAR  14444135   
3010848  10/18/2019  13:00:00     REGULAR  10994822   
3010847  10/18/2019  13:00:00     REGULAR  10994822   
3010846  10/18/2019  13:00:00     REGULAR   5100839   

         Exits                                                       \
3007920                                            1945485            
3010850                                            4540530            
3010851                                            3993107            
3010852                                           10975178            
3010853                                             925048            
3010854                                             925048            
3010855                                           10179811            
3010856                                           10179811            
3010857                                            3993107            
3010858                                           10975178            
3010859                                            4540530            
3010860                                            4755459            
3010861                                            4755459            
3010862                                             197583            
3010863                                           44108376            
3010864                                           44108376            
3010849                                           11509079            
3010848                                            6061830            
3010847             

In [26]:
#now order by unix timestamp and the name in alphabetical order

man_turn_data = man_turn_data.sort_values(by=['name','UnixTimestamp'])
man_turn_data.head(20)

name        lat        lon   C/A       SCP Station  \
3037515  103RD STREET  40.790298 -73.947624  R252  00-03-02  103 ST   
3037514  103RD STREET  40.790298 -73.947624  R252  00-00-01  103 ST   
3037513  103RD STREET  40.790298 -73.947624  R252  00-03-00  103 ST   
3037512  103RD STREET  40.790298 -73.947624  R252  00-03-01  103 ST   
3037511  103RD STREET  40.790298 -73.947624  R170  00-00-00  103 ST   
3037510  103RD STREET  40.790298 -73.947624  R170  00-00-02  103 ST   
3037509  103RD STREET  40.790298 -73.947624  R170  00-03-00  103 ST   
3037508  103RD STREET  40.790298 -73.947624  R170  00-03-02  103 ST   
3037507  103RD STREET  40.790298 -73.947624  R170  00-03-01  103 ST   
3037506  103RD STREET  40.790298 -73.947624  R170  00-00-01  103 ST   
3037516  103RD STREET  40.790298 -73.947624  R252  00-00-02  103 ST   
3037505  103RD STREET  40.790298 -73.947624  N037  00-00-02  103 ST   
3037503  103RD STREET  40.790298 -73.947624  N037  00-00-00  103 ST   
3037504  103RD STREET  40.790298 -73.947624  N037  00-00-01  103 ST   
3037517  103RD STREET  40.790298 -73.947624  R252  00-00-00  103 ST   
3037519  103RD STREET  40.790298 -73.947624  N037  00-00-01  103 ST   
3037532  103RD STREET  40.790298 -73.947624  R252  00-03-02  103 ST   
3037518  103RD STREET  40.790298 -73.947624  N037  00-00-02  103 ST   
3037531  103RD STREET  40.790298 -73.947624  R252  00-03-01  103 ST   
3037529  103RD STREET  40.790298 -73.947624  R252  00-00-01  103 ST   

               Date      Time Description   Entries  \
3037515  12/29/2018  00:00:00     REGULAR   8476658   
3037514  12/29/2018  00:00:00     REGULAR   3999244   
3037513  12/29/2018  00:00:00     REGULAR    408415   
3037512  12/29/2018  00:00:00     REGULAR   1659007   
3037511  12/29/2018  00:00:00     REGULAR   1634656   
3037510  12/29/2018  00:00:00     REGULAR  16113503   
3037509  12/29/2018  00:00:00     REGULAR   8149599   
3037508  12/29/2018  00:00:00     REGULAR  11186555   
3037507  12/29/2018  00:00:00     REGULAR  12358341   
3037506  12/29/2018  00:00:00     REGULAR  13143142   
3037516  12/29/2018  00:00:00     REGULAR  14397606   
3037505  12/29/2018  00:00:00     REGULAR   4888530   
3037503  12/29/2018  00:00:00     REGULAR  13856245   
3037504  12/29/2018  00:00:00     REGULAR  10577620   
3037517  12/29/2018  00:00:00     REGULAR  36496861   
3037519  12/29/2018  04:00:00     REGULAR  10577630   
3037532  12/29/2018  04:00:00     REGULAR   8476683   
3037518  12/29/2018  04:00:00     REGULAR   4888533   
3037531  12/29/2018  04:00:00     REGULAR   1659029   
3037529  12/29/2018  04:00:00     REGULAR   3999274   

         Exits                                                       \
3037515                                           26737489            
3037514                                            1607925            
3037513                                             557699            
3037512                                            3170302            
3037511                                             656635            
3037510                                            4435458            
3037509                                            4546826            
3037508                                           10491083            
3037507                                            9863905            
3037506                                            3857058            
3037516                                            5560657            
3037505                                            8281670            
3037503                                           11185013            
3037504                                            5871697            
3037517                                          505556313            
3037519                                            5871735            
3037532                                           26737597            
3037518                                            8281732            
3037531             

- needed a combination of C/A and SCP to differentiate each turnstile at each time for each day

In [27]:
man_turn_data.tail(50)

name        lat       lon    C/A       SCP        Station  \
924855  WTC CORTLANDT  40.711115 -74.01227   R109  03-00-02  WTC-CORTLANDT   
924854  WTC CORTLANDT  40.711115 -74.01227  R108A  05-03-00  WTC-CORTLANDT   
924853  WTC CORTLANDT  40.711115 -74.01227  R108A  05-00-01  WTC-CORTLANDT   
924852  WTC CORTLANDT  40.711115 -74.01227  R108A  05-00-00  WTC-CORTLANDT   
924851  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-04  WTC-CORTLANDT   
924850  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-01  WTC-CORTLANDT   
924849  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-02  WTC-CORTLANDT   
924848  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-00  WTC-CORTLANDT   
924847  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-03  WTC-CORTLANDT   
924828  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-04  WTC-CORTLANDT   
924827  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-03  WTC-CORTLANDT   
924825  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-00  WTC-CORTLANDT   
924824  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-02  WTC-CORTLANDT   
924823  WTC CORTLANDT  40.711115 -74.01227   R106  01-05-01  WTC-CORTLANDT   
924826  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-01  WTC-CORTLANDT   
924879  WTC CORTLANDT  40.711115 -74.01227  R107D  04-03-01  WTC-CORTLANDT   
924878  WTC CORTLANDT  40.711115 -74.01227  R107D  04-05-00  WTC-CORTLANDT   
924877  WTC CORTLANDT  40.711115 -74.01227  R107D  04-00-02  WTC-CORTLANDT   
924876  WTC CORTLANDT  40.711115 -74.01227  R107D  04-00-01  WTC-CORTLANDT   
924875  WTC CORTLANDT  40.711115 -74.01227  R107D  04-00-00  WTC-CORTLANDT   
924874  WTC CORTLANDT  40.711115 -74.01227  R107D  04-00-03  WTC-CORTLANDT   
924880  WTC CORTLANDT  40.711115 -74.01227  R107D  04-03-02  WTC-CORTLANDT   
924873  WTC CORTLANDT  40.711115 -74.01227  R107D  04-03-00  WTC-CORTLANDT   
924871  WTC CORTLANDT  40.711115 -74.01227   R107  00-00-00  WTC-CORTLANDT   
924870  WTC CORTLANDT  40.711115 -74.01227   R107  00-00-01  WTC-CORTLANDT   
924869  WTC CORTLANDT  40.711115 -74.01227   R107  00-05-01  WTC-CORTLANDT   
924868  WTC CORTLANDT  40.711115 -74.01227   R107  00-00-02  WTC-CORTLANDT   
924867  WTC CORTLANDT  40.711115 -74.01227   R107  00-05-00  WTC-CORTLANDT   
924866  WTC CORTLANDT  40.711115 -74.01227   R107  00-03-02  WTC-CORTLANDT   
924872  WTC CORTLANDT  40.711115 -74.01227   R107  00-03-00  WTC-CORTLANDT   
924865  WTC CORTLANDT  40.711115 -74.01227   R107  00-03-01  WTC-CORTLANDT   
924881  WTC CORTLANDT  40.711115 -74.01227  R107D  04-05-01  WTC-CORTLANDT   
924883  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-00  WTC-CORTLANDT   
924892  WTC CORTLANDT  40.711115 -74.01227   R109  03-00-00  WTC-CORTLANDT   
924882  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-04  WTC-CORTLANDT   
924891  WTC CORTLANDT  40.711115 -74.01227   R109  03-00-02  WTC-CORTLANDT   
924889  WTC CORTLANDT  40.711115 -74.01227  R108A  05-00-01  WTC-CORTLANDT   
924888  WTC CORTLANDT  40.711115 -74.01227  R108A  05-03-00  WTC-CORTLANDT   
924887  WTC CORTLANDT  40.711115 -74.01227  R108A  05-00-00  WTC-CORTLANDT   
924886  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-02  WTC-CORTLANDT   
924885  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-03  WTC-CORTLANDT   
924884  WTC CORTLANDT  40.711115 -74.01227   R108  02-00-01  WTC-CORTLANDT   
924890  WTC CORTLANDT  40.711115 -74.01227   R109  03-00-01  WTC-CORTLANDT   
924864  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-03  WTC-CORTLANDT   
924863  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-01  WTC-CORTLANDT   
924862  WTC CORTLANDT  40.711115 -74.01227   R106  01-05-01  WTC-CORTLANDT   
924861  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-04  WTC-CORTLANDT   
924860  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-00  WTC-CORTLANDT   
924859  WTC CORTLANDT  40.711115 -74.01227   R106  01-00-02  WTC-CORTLANDT   
924858  WTC CORTLANDT  40.711115 -74.01227   R106  01-05-00  WTC-CORTLANDT   

              Date      Time Descripti

In [28]:
#man_turn_data.to_csv('man_turn_data.csv')